patches/minecraft/net/minecraft/world/raid/Raid.java.patch

```diff
@@ -0,0 +1,56 @@
+--- a/net/minecraft/world/raid/Raid.java
++++ b/net/minecraft/world/raid/Raid.java
+@@ -201,12 +201,14 @@
+ 
+    public void func_221295_o() {
+       this.field_221353_q = false;
++      if (net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(new net.minecraftforge.event.world.RaidEvent.RaidStopEvent(this))) return;
+       this.field_221355_s.func_201360_b();
+       this.field_221360_x = Raid.Status.STOPPED;
+    }
+ 
+    public void func_221307_p() {
+       if (!this.func_221310_d()) {
++    	 if (net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(new net.minecraftforge.event.world.RaidEvent.RaidTickEvent(this))) return;
+          if (this.field_221360_x == Raid.Status.ONGOING) {
+             boolean flag = this.field_221353_q;
+             this.field_221353_q = this.field_221348_l.func_175667_e(this.field_221347_k);
+@@ -296,6 +298,7 @@
+             while(this.func_221318_F()) {
+                BlockPos blockpos = this.field_221362_z.isPresent() ? this.field_221362_z.get() : this.func_221298_a(k, 20);
+                if (blockpos != null) {
++            	  if (net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(new net.minecraftforge.event.world.RaidEvent.RaidStart(this)));
+                   this.field_221349_m = true;
+                   this.func_221294_b(blockpos);
+                   if (!flag3) {
+@@ -462,6 +465,7 @@
+ 
+          for(int l = 0; l < j; ++l) {
+             AbstractRaiderEntity abstractraiderentity = raid$wavemember.field_221285_g.func_200721_a(this.field_221348_l);
++            if (net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(new net.minecraftforge.event.world.RaidEvent.RaidSpawnRaidersEvent(this))) return;
```

Comment:
Why not include the entity in the event? Why is it called "SpawnRaidersEvent", when its really an event for when a single raider is spawned?
Also, there are more raiders spawned below in this method, why do they not fire the event?

Reply:
included the entity in the event, renamed the event, and made the getPotentialBonusSpawns fire that event